### PR TITLE
fix: run lint on complete project before commit

### PR DESCRIPTION
**Pull request by gupta-piyush19**
## What does this PR do?
Run lint on complete project before commit

Fixes
https://github.com/formbricks/internal/issues/258

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by gupta-piyush19** [comment]
@mattinannt should we remove `lint-staged`, as it's not being used anymore? 🙂

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-pnpm lint-staged
\ No newline at end of file
+pnpm lint
```
Comment (mattinannt) [inline .husky/pre-commit:4]: what about the other commands we are using in lint-staged, like prettier? are these now skipped?
Do we still need the lint-staged dependency?

**Review by mattinannt — CHANGES_REQUESTED** [review]

**Comment by mattinannt** [comment]
@gupta-piyush19 I think we can find a better solution for the problem. Now we introduce other issues that we aren't using prettier anymore, the linting command outputs a lot and takes much longer to complete. 

The main issue that I saw why we needed this PR is, that lint-staged just ran `eslint -fix` from the root directory doesn't seem to work correctly.
But maybe it's the better approach to get this working since lint-staged has some advantages regarding performance since only staged files are scanned and can also run the prettier command on all the staged files.

If we get `eslint` running so it takes the correct eslint config for the package/app and also work without errors, this should be the preferred approach.

Can you please research with the lint-staged docs and chatGPT how we can get the previous approach running?

**Review by mattinannt — APPROVED** [review]
@gupta-piyush19 nice, thanks for solving this :-) 💪
Now we can build great new linting pipelines 🚀

**Pull request merged**